packages/core/src/auth/sso/clients.ts

```diff
@@ -113,7 +113,7 @@ export class OidcClient {
             // InvalidRequestRegionException
             // InvalidRedirectUriException
             // InvalidRedirectUriException
-            return err.name === 'InvalidGrantException'
+            return false
```

Comment:
should the comment above this be updated?

> Do not retry on InvalidGrantException since there is no point. When we get that error, there is no scenario where things will work again. This was indicated to us by someone on the identity team.

That's surprising. This code was added in https://github.com/aws/aws-toolkit-vscode/pull/3498 to fix a sev2. Reviewing that commit (and related, see also https://sim.amazon.com/issues/IDE-10703 ), can we form an explanation that gives confidence that this won't cause a regression? Did this change not actually fix the sev2?

Reply:
Getting final verification again, but updated the ticket with references to previous conversations: https://sim.amazon.com/issues/IDE-10703?selectedConversation=e611a3c1-6924-477f-8b65-37638ee2a063

Reply:
Nice. This would be a great comment to add in the source code here:

> retry on invalid_grant would not be recommended as it's an indication that refresh token is invalid now.

Reply:
Updated the comments. It will now give more context and points to our SIM ticket